fhir-persistence/src/main/java/com/ibm/fhir/persistence/util/InputOutputByteStream.java

```diff
@@ -73,7 +73,7 @@ public int newSize(int currentSize, int desiredSize) {
         @Override
         public void write(int b) throws IOException {
             int idx = offset++;
-            extend(idx);
+            extend(offset); // use the new length
```

Comment:
this fixes a latent bug that has been here since the inception of this class...probably missed previously because the other write method was being used instead of this one

Reply:
Correct. Previous uses never called this method so didn't hit it.